controllers/front/ProductController.php

```diff
@@ -826,25 +836,30 @@ protected function formatQuantityDiscounts($specific_prices, $price, $tax_rate,
             if ($row['price'] >= 0) {
                 // The price may be directly set
 
-                $cur_price = (!$row['reduction_tax'] ? $row['price'] : $row['price'] * (1 + $tax_rate / 100)) + (float) $ecotax_amount;
+                /** @var float $currentPriceDefaultCurrency current price with taxes in default currency */
+                $currentPriceDefaultCurrency = (!$row['reduction_tax'] ? $row['price'] : $row['price'] * (1 + $tax_rate / 100)) + (float) $ecotax_amount;
```

Comment:
Try using [`PrestaShop\Decimal\Number`](https://github.com/PrestaShop/decimal), we should start using it everywhere, especially when in places where we perform currency divisions.

Reply:
I think this is out of current scope, here I simply renamed the var  with a more meaningful name

Reply:
After discussion, we'll do that later